product/consul.go

```diff
@@ -26,8 +28,10 @@ func NewConsul(cfg Config) (*Product, error) {
 		return nil, err
 	}
 	return &Product{
+		l:       logger.Named("product"),
```

Comment:
I'm not super familiar with the hclog library, but I wonder, if we're creating new named loggers for each product, should we also indicate the product type? For example, `"product.consul"`? This is just complete speculation/curiosity on my part.

Reply:
I tried it both ways. hclog convention is to name by subsystem which is more "product" than each individual product. We also print out the product name in logs down the line so the info is available. One way we could improve that though is storing the named logger `.With(product.Name)` which ensures that key is printed in every logline, much like we do now by hand in the line.